turbolinks/src/main/java/com/basecamp/turbolinks/TurbolinksView.java

```diff
@@ -99,21 +110,26 @@ public void run() {
     }
 
     /**
-     * <p>Removes the progressView from the TurbolinksView. Ensures no exceptions are thrown where
-     * the progressView is already attached to another view.</p>
+     * <p>Removes the progress view and/or screenshot from the TurbolinksView, so the webview is
+     * visible underneath.</p>
      */
-    void removeProgressView() {
-        removeView(progressView);
+    void hideProgress() {
+        removeProgressView();
+        removeScreenshotView();
     }
 
     /**
      * <p>Attach the shared webView to the TurbolinksView.</p>
      *
      * @param webView The shared webView.
+     * @param screenshotsEnabled Indicates whether screenshots are enabled for the current session.
      */
-    void attachWebView(WebView webView) {
-        ViewGroup parent = (ViewGroup) webView.getParent();
-        if (parent != null) {
+    void attachWebView(WebView webView, boolean screenshotsEnabled) {
+        if (webView.getParent() == this) return;
+
+        if (webView.getParent() instanceof TurbolinksView) {
```

Comment:
There shouldn't be a valid reason why the parent is anything other than a `TurbolinksView` or `null`. I decided to combine the sanity and `null` check into this single statement, since `instanceof` also `null` checks.